urlddedf61e88_dwendelen_CG2_1_tgz-pJ8-RendProjectJ8/src/ImagePack/RGBEFormat.java

```diff
@@ -0,0 +1,116 @@
+package ImagePack;
+
+
+import java.io.*;
+
+import Color.*;
+
+
+
+/**
+ * Support for uncompressed RGBE image in/out.
+ * @author  fabio
+ * @version 0.1
+ */
+
+public abstract class RGBEFormat {
+  private static int			width, height;
+  private static float[]		rgbData;
+  private static byte[]			rgbeData;
+  private static Double			gamma;
+  private static Double			exposure;
+
+  // write file
+  public static synchronized void write(Image image, Double gamma,
+                                        Double exposure, String filename)
+                                        throws IOException {
+    setData(image,gamma,exposure);
+    OutputStream os = new BufferedOutputStream(new FileOutputStream(filename), 8192);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *os*. Execution paths that do not contain closure statements were detected. Ensure *os* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).